### PR TITLE
Let the dep-tracking test framework check that all #[rustc_dirty] attrs have been actually checked

### PR DESCRIPTION
**Pull request by michaelwoerister**
r? @nikomatsakis

### CODE REVIEW

**Comment by michaelwoerister** [comment]
Before r+ing, please wait for travis to finish as I haven't had the time to run a full `make check` and that change to the visitor might have a wider effect.

File: src/librustc/hir/intravisit.rs [hunk 1]
```diff
@@ -914,6 +914,7 @@ pub fn walk_decl<'v, V: Visitor<'v>>(visitor: &mut V, declaration: &'v Decl) {
 
 pub fn walk_expr<'v, V: Visitor<'v>>(visitor: &mut V, expression: &'v Expr) {
     visitor.visit_id(expression.id);
+    walk_list!(visitor, visit_attribute, expression.attrs.iter());
```
Comment (nikomatsakis) [inline src/librustc/hir/intravisit.rs]: d'oh

File: src/librustc_incremental/persist/dirty_clean.rs [hunk 2]
```diff
@@ -73,17 +74,29 @@ pub fn check_dirty_clean_annotations<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
...
+    krate.visit_all_item_likes(&mut dirty_clean_visitor);
+
+    let mut all_attrs = FindAllAttrs {
+        tcx: tcx,
+        attr_names: vec![ATTR_DIRTY, ATTR_CLEAN],
+        found_attrs: vec![],
+    };
+    intravisit::walk_crate(&mut all_attrs, krate);
+
+    all_attrs.report_unchecked_attrs(&dirty_clean_visitor.checked_attrs);
```
Comment (nikomatsakis) [inline src/librustc_incremental/persist/dirty_clean.rs]: can we add a comment explaining why we are not using the other "used attribute" system?

**Review by nikomatsakis — APPROVED** [review]
r=me with comment

**Comment by michaelwoerister** [comment]
@bors r=nikomatsakis

(comments added)

**Pull request merged**